src/main/java/uk/gov/hmcts/sptribs/common/event/CreateTestCase.java

```diff
@@ -129,23 +136,73 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<C
     public SubmittedCallbackResponse submitted(CaseDetails<CaseData, State> details,
                                                CaseDetails<CaseData, State> beforeDetails) {
         var data = details.getData();
-        String claimNumber = data.getHyphenatedCaseRef();
 
-        sendApplicationReceivedNotification(claimNumber, data);
+        setSupplementaryData(details.getId());
+
+        String claimNumber = data.getHyphenatedCaseRef();
+        try {
+            sendApplicationReceivedNotification(claimNumber, data);
+        } catch (Exception notificationException) {
+            log.error("Create case notification failed with exception : {}", notificationException.getMessage());
+            return SubmittedCallbackResponse.builder()
+                .confirmationHeader(format("# Create case notification failed %n## Please resend the notification"))
+                .build();
+        }
 
         return SubmittedCallbackResponse.builder()
             .confirmationHeader(format("# Case Created %n## Case reference number: %n## %s", claimNumber))
             .build();
     }
 
+    private void initialiseFlags(CaseData data) {
+        data.setCaseFlags(Flags.builder()
+            .details(new ArrayList<>())
+            .partyName(null)
+            .roleOnCase(null)
+            .build());
+
+        data.setSubjectFlags(Flags.builder()
+            .details(new ArrayList<>())
+            .partyName(data.getCicCase().getFullName())
+            .roleOnCase("subject")
+            .build()
+        );
+
+        if (null != data.getCicCase().getApplicantFullName()) {
+            data.setApplicantFlags(Flags.builder()
+                .details(new ArrayList<>())
+                .partyName(data.getCicCase().getApplicantFullName())
+                .roleOnCase("applicant")
+                .build()
+            );
+        }
+
+        if (null != data.getCicCase().getRepresentativeFullName()) {
+            data.setRepresentativeFlags(Flags.builder()
+                .details(new ArrayList<>())
+                .partyName(data.getCicCase().getRepresentativeFullName())
+                .roleOnCase("Representative")
+                .build()
+            );
+        }
+    }
+
+    private void setSupplementaryData(Long caseId) {
+        try {
+            coreCaseApiService.submitSupplementaryDataToCcd(caseId.toString());
+        } catch (Exception exception) {
+            log.error("Unable to set Supplementary data with exception : {}", exception.getMessage());
+        }
+    }
+
     private void sendApplicationReceivedNotification(String caseNumber, CaseData data) {
         CicCase cicCase = data.getCicCase();
 
         if (CollectionUtils.isNotEmpty(cicCase.getSubjectCIC())) {
             applicationReceivedNotification.sendToSubject(data, caseNumber);
         }
 
-        if (CollectionUtils.isNotEmpty(cicCase.getApplicantCIC())) {
+        if (!cicCase.getApplicantCIC().isEmpty()) {
```

Comment:
Null check is implicitly included in CollectionUtils.isNotEmpty method. Why do you want to avoid null check?

Reply:
Reverted this change.